client/src/form/DropDownListBar.js

```diff
@@ -57,14 +65,28 @@ export default function BasicTextFields() {
             setErrMsg({open:true,errMsg:"ERROR: List is required."})
             return
         }
-        
-        addNewProduct(itemData.listId,shListsContext.dispatchNewProduct,{url:itemData.url});
-        setItemData({
-            listId: '',
-            url: '',
-        });
+        setLoading(true);
     }
 
+    useEffect(() => {
+        let isUnmounted = false;
+        if(loading) {
+            let promise = addNewProduct(itemData.listId,shListsContext.dispatchNewProduct,{url:itemData.url});
+            promise.then(() => {
+                if(!isUnmounted) {
+                    setLoading(false);
+                    setItemData({
+                        listId: '',
+                        url: '',
+                    });
+                }
+            });
+        }
+        return () => {
+            isUnmounted = true;
+        };
+    }, [loading]);
```

Comment:
I'm a little confused about the `isUnmounted` cleanup logic. Can you explain the thinking here?

Reply:
So basically I was searching for the error on stackoverflow and reading how to use useEffects in React docs. So I found a code - https://codesandbox.io/s/use-local-variable-to-avoid-setstate-on-unmouted-component-6cmie where a similar logic of isCancelled and setMounted was added. I followed the code and wrote this and it worked somehow!!

Reply:
So as far as I understood, we are calling a promise onSubmit and also updating the state (setLoading) so we need to move this code to useEffect and check if the component is mounted or not (something called as clean up function), a temp flag which returns true if the component is unmounted before the promise completes. Is that right?!